eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/JavadocCodeSnippetCheck.java

```diff
@@ -0,0 +1,194 @@
+                // Checkstyle complains if there's no default block in switch
+                break;
+        }
+    }
+
+    /**
+     * Requirement of rules:
+     * (1) No usage of '<code>', '<pre>',  use '{@codesnippet ...} instead'.
+     * (2) No multiple lines span at the '{@code ...}'
+     *
+     * @param htmlElementStartNode
+     */
+    private void checkHtmlElementStart(DetailNode htmlElementStartNode) {
+        final DetailNode tagNameNode = JavadocUtil.findFirstToken(htmlElementStartNode, JavadocTokenTypes.HTML_TAG_NAME);
+        final String tagName = tagNameNode.getText();
+        if ("code".equals(tagName) || "pre".equals(tagName)) {
+            log(tagNameNode.getLineNumber(), tagNameNode.getColumnNumber(),
+                String.format("Do not use <%s> html tag in javadoc. Use <codesnippet> instead.", tagName));
+        }
+    }
+
+    /**
+     * Check to see if the JAVADOC_INLINE_TAG node is '@code' tag or '@codesnippet' tag.
+     * If the JAVADOC_INLINE_TAG is one of two, check if the tag contains new line or leading asterisk,
+     * which implies the tag has spanned in multiple lines.
+     *
+     * @param inlineTagNode JAVADOC_INLINE_TAG javadoc node
+     */
+    private void checkJavadocInlineTag(DetailNode inlineTagNode) {
+        boolean isCodeOrCodeSnippet = false;
+        if (JavadocUtil.findFirstToken(inlineTagNode, JavadocTokenTypes.CODE_LITERAL) != null) {
+            isCodeOrCodeSnippet = true;
+        }
+        isCodeOrCodeSnippet |= isCodeSnippet(inlineTagNode);
+        if (!isCodeOrCodeSnippet) {
+            return;
+        }
+
+        for (final DetailNode child : inlineTagNode.getChildren()) {
+            final int childType = child.getType();
+
+            if (childType == JavadocTokenTypes.NEWLINE || childType == JavadocTokenTypes.LEADING_ASTERISK) {
+                log(child.getLineNumber(), child.getColumnNumber(), "No multiple lines in @code annotation");
+            }
+
+            // This code section duplicates the checking on @codesnippet, maven build already failed it
+            if (childType == JavadocTokenTypes.DESCRIPTION
+                && (!JavadocUtil.containsInBranch(child, JavadocTokenTypes.NEWLINE)
+                    || !JavadocUtil.containsInBranch(child, JavadocTokenTypes.LEADING_ASTERISK))) {
+                log(child.getLineNumber(), child.getColumnNumber(), "No multiple lines in @codesnippet annotation");
+            }
+        }
+        return;
+    }
+
+    /**
+     * Check to see if the code snippet name matched the naming pattern.
+     *
+     * @param inlineTagNode JAVADOC_INLINE_TAG javadoc node
+     */
+    private void checkCodeSnippetNaming(DetailNode inlineTagNode) {
+        if (!isCodeSnippet(inlineTagNode)) {
+            return;
+        }
+
+        final DetailNode descriptionNode = JavadocUtil.findFirstToken(inlineTagNode, JavadocTokenTypes.DESCRIPTION);
+        if (descriptionNode == null) {
+            return;
```

Comment:
Shouldn't it be an error if the `@codesnippet` node doesn't have a description nor any text?? 🤔